api/http/handler/registries/registry_delete.go

```diff
@@ -50,5 +52,55 @@ func (handler *Handler) registryDelete(w http.ResponseWriter, r *http.Request) *
 		return httperror.InternalServerError("Unable to remove the registry from the database", err)
 	}
 
+	err = handler.deleteKubernetesSecrets(registry)
+	if err != nil {
+		return httperror.InternalServerError("Unable to delete registry secrets", err)
+	}
+
 	return response.Empty(w)
 }
+
+func (handler *Handler) deleteKubernetesSecrets(registry *portainer.Registry) error {
+
+	for endpointId, access := range registry.RegistryAccesses {
+		if access.Namespaces != nil {
+			// Obtain a kubeclient for the endpoint
+			endpoint, err := handler.DataStore.Endpoint().Endpoint(endpointId)
+			if err != nil {
+				// Skip environments that can't be loaded. Log an error unless they don't exist.
+				if !handler.DataStore.IsErrObjectNotFound(err) {
+					log.Warn().Err(err).Msgf("Unable to load the environment with id %d from the database", endpointId)
+				}
```

Comment:
Why not log all errors?

Reply:
Because at the time I figured who cares if we have some out of date data.  Just move along.  But it's a fair comment.